### ISSUE TITLE
Re-open editor with unsaved documents open to much windows

### ISSUE THREAD

**Issue by DoCode** [Microsoft/vscode#60475]
Issue Type: <b>Bug</b>

The editor opens multiple windows when one or multiple unsaved documents are opened in the last session.

VS Code version: Code - Insiders 1.29.0-insider (cc5e2b41c3add18a31e87f84b5d6162566a38d6e, 2018-10-10T05:19:52.883Z)
OS version: Windows_NT x64 10.0.17763

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-5930K CPU @ 3.50GHz (12 x 3566)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|31.93GB (23.04GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (31)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-file-peek|abi|1.0.1
pascal|ale|6.4.0
markdown-checkbox|bie|0.1.1
markdown-mermaid|bie|1.0.0
vscode-eslint|dba|1.6.0
vscode-new-file|dku|4.0.2
docs-markdown|doc|0.1.5
xml|Dot|2.3.2
gitlens|eam|8.5.6
LogFileHighlighter|emi|2.4.2
prettier-vscode|esb|1.6.1
auto-close-tag|for|0.5.6
graphviz-markdown-preview|gee|0.0.7
live-html-previewer|hdg|0.3.0
beautify|Hoo|1.4.4
rest-client|hum|0.20.2
innosetup|idl|1.2.0
reg|ion|0.0.2
python|ms-|2018.9.0
csharp|ms-|1.16.2
Go|ms-|0.6.90
PowerShell|ms-|1.9.0
vetur|oct|0.13.0
vscode-docker|Pet|0.3.1
vscode-sort-json|ric|1.13.0
vscode-icons|rob|7.27.0
ActiveFileInStatusBar|Ros|1.0.3
trailing-spaces|sha|0.2.11
vscode-autohotkey|sle|0.2.2
vscode-hexdump|sle|1.7.0
markdown-all-in-one|yzh|1.6.2

</details>

**Comment by bpasero** [comment]
@DoCode can you provide exact steps how to reproduce. Make sure you start from a blank state, e.g. no unsaved documents.

**Comment by DoCode** [comment]
1. Open VS Code
2. Close all tabs, also the welcome one
3. Open a new tab
4. Insert some text
5. Close VS Code without saving the file
6. Re-open the VS Code
7. Now, two VS Code windows are opened, the unsaved file is opened in each window!

**Comment by Tyriar** [comment]
Can you launch VS Code using `code --disable-extensions` and also share your settings.json file?

**Comment by DoCode** [comment]
1) Launch VS Code with `code --disable-extensions` results with the same issue!

2) Here is my `settings.json`:
```json
{
  "ActiveFileInStatusBar.color": "lime",
  "ActiveFileInStatusBar.enable": true,
  "ActiveFileInStatusBar.fullpath": true,
  "ActiveFileInStatusBar.revealFile": false,
  "[bat]": {
    "editor.tabSize": 2
  },
  "[git-commit]": {
    "editor.tabSize": 2
  },
  "[git-rebase]": {
    "editor.tabSize": 2
  },
  "[innopascal]": {
    "editor.tabSize": 2
  },
  "[javascript]": {
    "editor.tabSize": 4
  },
  "[json]": {
    "editor.tabSize": 2
  },
  "[markdown]": {
    "editor.tabSize": 4
  },
  "[pascal]": {
    "editor.tabSize": 2
  },
  "[plaintext]": {
    "editor.tabSize": 2
  },
  "[powershell]": {
    "editor.tabSize": 4
  },
  "[python]": {
    "editor.tabSize": 4
  },
  "[xml]": {
    "editor.tabSize": 2
  },
  "beautify.language": {
    "css": ["css", "scss"],
    "html": ["htm", "html"],
    "js": {
      "filename": [".jshintrc", ".jsbeautify"],
      "type": ["javascript", "json"]
    }
  },
  "breadcrumbs.enabled": true,
  "editor.accessibilitySupport": "off",
  "editor.cursorStyle": "block",
  "editor.cursorWidth": 10,
  "editor.detectIndentation": true,
  "editor.fontSize": 13.5,
  "editor.lineNumbers": "on",
  "editor.links": false,
  "editor.minimap.enabled": true,
  "editor.renderLineHighlight": "line",
  "editor.renderWhitespace": "boundary",
  "editor.stablePeek": true,
  "eslint.enable": true,
  "eslint.options": {
    "rules": {
      "array-bracket-spacing": [
        "error",
        "always",
        {
          "objectsInArrays": false
        }
      ]
    }
  },
  "explorer.confirmDelete": false,
  "explorer.confirmDragAndDrop": false,
  "files.hotExit": "onExitAndWindowClose",
  "files.trimFinalNewlines": true,
  "files.trimTrailingWhitespace": true,
  "gitlens.advanced.messages": {
    "suppressCommitHasNoPreviousCommitWarning": false,
    "suppressCommitNotFoundWarning": false,
    "suppressFileNotUnderSourceControlWarning": false,
    "suppressGitVersionWarning": false,
    "suppressLineUncommittedWarning": false,
    "suppressNoRepositoryWarning": false,
    "suppressResultsExplorerNotice": false,
    "suppressShowKeyBindingsNotice": true
  },
  "gitlens.historyExplorer.enabled": true,
  "gitlens.keymap": "none",
  "hexdump.width": 32,
  "json.schemas": [
    {
      "fileMatch": ["config.json"],
      "url": "http://jsonschema.azurewebsites.net/schemas/config"
    }
  ],
  "markdown.extension.preview.autoShowPreviewToSide": false,
  "markdown.extension.toc.githubCompatibility": true,
  "markdown.extension.toc.unorderedList.marker": "-",
  "markdown.extension.toc.updateOnSave": false,
  "markdown.preview.markEditorSelection": true,
  "markdown.preview.scrollEditorWithPreview": true,
  "markdown.preview.scrollPreviewWithEditor": true,
  "markdown.styles": [],
  "powershell.codeFormatting.openBraceOnSameLine": true,
  "powershell.codeFormatting.preset": "OTBS",
  "powershell.codeFormatting.whitespaceBeforeOpenParen": true,
  "powershell.integratedConsole.showOnStartup": false,
  "powershell.powerShellExePath": "C:\\Program Files\\PowerShell\\6-preview\\pwsh.exe",
  "prettier.eslintIntegration": true,
  "prettier.printWidth": 120,
  "prettier.singleQuote": true,
  "telemetry.enableCrashReporter": false,
  "telemetry.enableTelemetry": false,
  "trailing-spaces.saveAfterTrim": true,
  "trailing-spaces.syntaxIgnore": ["markdown"],
  "trailing-spaces.trimOnSave": true,
  "window.closeWhenEmpty": false,
  "window.menuBarVisibility": "toggle",
  "window.newWindowDimensions": "default",
  "window.restoreWindows": "all",
  "window.title": "${dirty}${activeEditorMedium}${separator}${rootName}${separator}${appName}",
  "window.titleBarStyle": "custom",
  "window.zoomLevel": 0,
  "workbench.activityBar.visible": true,
  "workbench.editor.closeOnFileDelete": false,
  "workbench.editor.enablePreview": true,
  "workbench.editor.enablePreviewFromQuickOpen": true,
  "workbench.editor.openPositioning": "last",
  "workbench.editor.showTabs": true,
  "workbench.iconTheme": "vscode-icons",
  "workbench.list.openMode": "singleClick",
  "workbench.sideBar.location": "right",
  "workbench.startupEditor": "welcomePage",
  "xmlTools.enableXmlTreeViewCursorSync": false
}
```

**Comment by DoCode** [comment]
Now the same with the latest build!

**Comment by DoCode** [comment]
@Tyriar now that's the same on the stable channel: Always when unsaved files are present and the edit is closed, all VS Code windows doubled on the next start!

**Comment by Tyriar** [comment]
@DoCode see https://github.com/Microsoft/vscode/issues/70006

**Comment by DoCode** [comment]
@Tyriar ok I can confirm that it was fixed in the latest 1.32.2 release
